tests/mshevchuk/homework_6.py

```diff
@@ -0,0 +1,76 @@
+from operator import add, mul
+
+
+def check_args(args):
+    if not args:
+        raise ValueError("arg is an empty sequence")
+
+
+def min(*args, key=None):
+    check_args(args)
+    key = key or (lambda i: i)
+    min_value = args[0]
+    for i in args[1:]:
+        if key(i) < key(min_value):
+            min_value = i
+    return min_value
+
+
+def max(*args, key=None):
+    check_args(args)
+    key = key or (lambda i: i)
+    max_value = args[0]
+    for i in args[1:]:
+        if key(i) > key(max_value):
+            max_value = i
+    return max_value
+
+
+def sorted(*args, key=None, reverse=False):
+    """Ascending by default"""
+    check_args(args)
+    func = max if reverse else min
+    args = list(args)
+    new_list = list()
+    while args:
+        value = func(*args, key=key)
+        new_list.append(value)
+        args.remove(value)
+    return new_list
+
+
+def reduce(*args, key: callable, default):
+    if args:
+        key = key or (lambda i: i)
+        value = args[0]
+        for i in args[1:]:
+            value = key(value, i)
+        return value
+    else:
+        return default
+
+
+def sum(*args):
+    check_args(args)
+    sum = 0
+    for i in args:
+        sum += i
+    return sum
```

Comment:
reuse reduce here

Reply:
sum is a bad name for var as it is name of function, alsoo we have builtin function sum